handler.js

```diff
@@ -360,117 +370,230 @@ class GithubBuild {
+        const {forceType, skipDeployment} = this.dataSet;
+        const checkForceTypes = ["deployment", "functional", "uat"];
+        return forceType && !skipDeployment && checkForceTypes.includes(forceType);
+      },
+      getTests: () => {
+        return [{
+          name: "deployment",
+          type: "deployment",
+          deployable: true
+        }];
       }
+    }
+  };
+
+ checkForceCommandForceArgument() {
+   return {
+     isMatch: () => {
+       const {forceArgument} = this.dataSet;
+       return forceArgument && typeof forceArgument !== "undefined";
+     },
+     getTests: () => {
+       return [{
+         name: this.dataSet.forceArgument.trim(),
+         type: this.dataSet.forceType,
+         deployable: false
+       }];
+     }
+   }
+  };
 
-      // check if there is a forcecommand and if we have enabled them
-      if(forceCommand && this.enableForceUATCommands()) {
-        const forceType = forceCommand[0];
-        const forceArgument = forceCommand[1];
-
-        if(typeof forceArgument !== "undefined") {
-          tests.push({
-            name: forceArgument.trim(),
-            type: forceType,
-            deployable: !skipDeployment
-          });
-        } else if (forceType === "uat") {
-          tests.push({
+  checkForceCommandUat() {
+    return {
+      isMatch: () => {
+        const {forceArgument, forceType} = this.dataSet;
+        return typeof forceArgument === "undefined" && forceType === "uat";
+      },
+      getTests: () => {
+        return [
+          {
             name: "backend",
             type: "uat",
-            deployable: !skipDeployment
-          });
-
-          tests.push({
+            deployable: false
+          },
+          {
             name: "frontend",
             type: "uat",
             deployable: false
-          });
-
-          tests.push({
+          },
+          {
             name: "functional",
             type: "functional",
             deployable: false
-          });
-        } else if (forceType === "functional") {
-          tests.push({
-            name: "functional",
-            type: "functional",
-            deployable: !skipDeployment
-          });
-        }
+          }
+        ];
+      }
+    }
+  };
+
+  checkForceCommandFunctional() {
+    return {
+      isMatch: () => {
+        const {forceArgument, forceType} = this.dataSet;
+        return typeof forceArgument === "undefined" && forceType === "functional";
+      },
+      getTests: () => {
+        return [{
+          name: "functional",
+          type: "functional",
+          deployable: false
+        }];
+      }
+    }
+  };
+
+  // combine all objects into a single array if some of the objects are in a sub-array
+  prepareArray(arr){
+    // Ref: https://www.jstips.co/en/javascript/flattening-multidimensional-arrays-in-javascript/
+    return [].concat.apply([], arr);
+  }
+
+  // return all the force commands related to test
+  checkForForceCommands() {
```

Comment:
This should be flattened out.

Reply:
Can you please let me know what did you mean by "flattened out" @tuanmh? Remove condition checks?

Reply:
yes, I don't think it's a good idea to include rules within rule.